jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -1614,6 +1614,110 @@ ecma_op_typedarray_define_index_prop (ecma_object_t *obj_p, /**< a TypedArray ob
   return ECMA_VALUE_TRUE;
 } /* ecma_op_typedarray_define_index_prop */
 
+/* specify the creation of a new TypedArray
+ * object using a constructor function.
+ *
+ * See also: ES11 22.2.4.6
+ *
+ * Used by:
+ *        - ecma_typedarray_species_create
+ *
+ * @return ecma_value_t
+ */
+ecma_value_t ecma_typedarray_create (ecma_object_t *constructor_p, /**< constructor function */
+                                     ecma_value_t *arguments_list_p, /**< argument list */
+                                     uint32_t arguments_list_len) /**< length of argument list */
+{
+  ecma_number_t num;
+  if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (arguments_list_p[0], &num)))
+  {
+    return ECMA_VALUE_ERROR;
+  }
+  uint32_t length = ecma_number_to_uint32 (num);
+
+  // 1. Let newTypedArray be ? Construct(constructor, argumentList).
+  ecma_value_t ret_val = ecma_op_function_construct (constructor_p,
+                                                     constructor_p,
+                                                     arguments_list_p,
+                                                     length);
+  // 2. Perform ? ValidateTypedArray(newTypedArray).
+  if (!ecma_is_typedarray (ret_val))
+  {
+    ecma_free_value (ret_val);
```

Comment:
`ret_val` can be `ECMA_VALUE_ERROR` at this point so you should create an early return after `Construct` call.